test/integration/emulator/FakeVideoDecoder.h

```diff
@@ -0,0 +1,244 @@
+    {
+        static constexpr size_t VP8_HEADER_SIZE = 2;
+        assert(length >= sizeof(fakenet::FakeVideoFrameData) + VP8_HEADER_SIZE);
+        const auto data = reinterpret_cast<const fakenet::FakeVideoFrameData*>(packet + VP8_HEADER_SIZE);
+
+        if (data->ssrc == 0)
+        {
+            assert(0); // Padding packets should have been dropped already in the RTP Receiver.
+            return;
+        }
+
+        _stats.numReceivedPackets++;
+
+        FAKE_VIDEO_DECODER_LOG(
+            "process packet for frame: %d, from %zu, ### last packet %c, key frame: %c (cache size = %zu)",
+            _loggableId.c_str(),
+            data->frameNum,
+            data->endpointIdHash,
+            data->lastPacketInFrame ? 't' : 'f',
+            data->keyFrame ? 't' : 'f',
+            _packets.size());
+
+        if (assembleFrame(*data))
+        {
+            removeDecodedPackets(data->frameNum);
+        }
+        else
+        {
+            FAKE_VIDEO_DECODER_LOG("putting packet to cache, frame: %d, from %zu, ###",
+                _loggableId.c_str(),
+                data->frameNum,
+                data->endpointIdHash);
+
+            _packets.push_back(*data);
+        }
+
+        decodeAssembledFrames(timestamp);
+    }
+
+    void resetPacketCache()
+    {
+        FAKE_VIDEO_DECODER_LOG("Reset packet cache (size = %zu)", _loggableId.c_str(), _packets.size());
+        _packets.clear();
+    }
+
+    Stats getStats() const { return _stats; }
+
+private:
+    void decodeAssembledFrames(const uint64_t timestamp)
+    {
+        while (_assembledFrames.size())
+        {
+            auto frameNum = _lastDecodedFrameNum + 1;
+            auto nextFrame = std::find_if(_assembledFrames.begin(),
+                _assembledFrames.end(),
+                [frameNum](const AssembledFrame& frame) { return frame.frameNum == frameNum || frame.keyFrame; });
+
+            if (nextFrame != _assembledFrames.end())
+            {
+                FAKE_VIDEO_DECODER_LOG("decoded frame: %d, from %zu",
+                    _loggableId.c_str(),
+                    nextFrame->frameNum,
+                    nextFrame->endpointIdHash);
+
+                _lastDecodedFrameNum = nextFrame->frameNum;
+                _ssrc = nextFrame->ssrc;
+
+                auto ssrc = _ssrc;
+                auto reorderQueueSize = std::count_if(_assembledFrames.begin(),
+                    _assembledFrames.end(),
+                    [frameNum, ssrc](
+                        const AssembledFrame& frame) { return (frame.frameNum > frameNum) && (ssrc == frame.ssrc); });
+
+                updateStats(nextFrame->endpointIdHash, nextFrame->tag, reorderQueueSize, timestamp);
+
+                _assembledFrames.remove_if([nextFrame](const AssembledFrame& frame) {
+                    return frame.ssrc != nextFrame->ssrc || frame.frameNum <= nextFrame->frameNum;
+                });
+            }
+            else
+            {
+                break;
```

Comment:
[minor] Please move break to close to the main condition, I mean inverse the condition on If-statement and break there. So it easier to understand the precondition fot the rest to run.
You will not need an "else" either